logger/logger_test.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2024 OWASP ModSecurity Core Rule Set Project
+// SPDX-License-Identifier: Apache-2.0
+
+package logger
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/suite"
+)
+
+type loggerTestSuite struct {
+	suite.Suite
+	out    *bytes.Buffer
+	logger zerolog.Logger
+}
+
+func TestRunLoggerTestSuite(t *testing.T) {
+	suite.Run(t, new(loggerTestSuite))
+}
+
+var testJsonBase = []struct {
+	name    string
+	text    string
+	logType zerolog.Level
+	want    string
+}{
+	{
+		name:    "JsonBaseOutput",
+		text:    "hello",
+		logType: zerolog.InfoLevel,
+		want:    "message\":\"hello\"",
+	},
+}
+
+var testConsoleBase = []struct {
+	name    string
+	text    string
+	logType zerolog.Level
+	want    string
+}{
+	{
+		name:    "BaseConsoleOutput",
+		text:    "hello",
+		logType: zerolog.InfoLevel,
+		want:    "INF hello component=parser-test",
+	},
+}
+
+var testGithub = []struct {
+	name    string
+	text    string
+	logType zerolog.Level
+	want    string
+}{
+	{
+		name:    "TestGithubInfoOutput",
+		text:    "this is an info message",
+		logType: zerolog.InfoLevel,
+		want:    "::notice ::this is an info message",
+	},
+	{
+		name:    "TestGithubWarningOutput",
+		text:    "this is a warning message",
+		logType: zerolog.WarnLevel,
+		want:    "::warn ::this is a warning message",
+	},
+	{
+		name:    "TestGithubTraceOutput",
+		text:    "this is a trace message that will show as debug",
+		logType: zerolog.TraceLevel,
+		want:    "::debug ::this is a trace message that will show as debug",
+	},
+	{
+		name:    "TestGithubDebugOutput",
+		text:    "this is a debug message",
+		logType: zerolog.DebugLevel,
+		want:    "::debug ::this is a debug message",
+	},
+}
+
+func (s *loggerTestSuite) SetupTest() {
+	// reset logger
+	s.out = &bytes.Buffer{}
+	s.logger = zerolog.New(s.out).With().Str("component", "parser-test").Logger()
+	zerolog.SetGlobalLevel(zerolog.TraceLevel)
+}
+
+func (s *loggerTestSuite) TestJsonOutput() {
+	for _, t := range testJsonBase {
+		s.Run(t.name, func() {
+			s.logger.WithLevel(t.logType).Msg(t.text)
+			s.Contains(s.out.String(), t.want)
+			s.out.Reset()
+		})
+	}
+}
+
+func (s *loggerTestSuite) TestConsoleOutput() {
+	s.logger = s.logger.Output(zerolog.ConsoleWriter{Out: s.out, NoColor: true, TimeFormat: "03:04:05"})
```

Comment:
Instead of configuring the console output here explicitly for the test, wouldn't it be better to have an additional choice to generate JSON output? Just like you did now for GH.

Reply:
Did you mean console output? JSON is the default. Do you want something like `SetConsoleOutput`?

Reply:
Zerolog's default is JSON, but our global logger is configured to use the console writer. So for users, currently, there's only the option of either console or GH optimised output. I don't see the point in testing JSON output if we don't offer it to users.